kernel/kernel-api/src/main/java/io/delta/kernel/expressions/CollationIdentifier.java

```diff
@@ -51,4 +51,23 @@ public static CollationIdentifier fromString(String identifier) {
       return new CollationIdentifier(parts[0], parts[1], Optional.of(parts[2]));
     }
   }
+
+  @Override
+  public String toString() {
+    String toString = toStringWithoutVersion();
```

Comment:
can we avoid calling this function if the version is actually present? Not a big deal but just seems off